onnxruntime/core/providers/cuda/cuda_execution_provider.cc

```diff
@@ -2532,9 +2532,8 @@ void CUDAExecutionProvider::RegisterStreamHandlers(IStreamCommandHandleRegistry&
 }
 
 OrtDevice CUDAExecutionProvider::GetOrtDeviceByMemType(OrtMemType mem_type) const {
-  if (mem_type == OrtMemTypeCPUInput || mem_type == OrtMemTypeCPUOutput) {
-    return OrtDevice(OrtDevice::CPU, OrtDevice::MemType::CUDA_PINNED, 0 /*CPU device id always be 0*/);
-  }
+  if (mem_type == OrtMemTypeCPUInput) return OrtDevice();
```

Comment:
>if (mem_type == OrtMemTypeCPUInput) return OrtDevice(); [](http://example.com/codeflow?start=2&length=55)

could you put some comments here why for CPU input we use non-pinned memory?

Reply:
Will do in the next PR